### PR TITLE
drop deprecated parameters from vectorize.table()

### LINKED ISSUE #146: drop deprecated parameters from `vectorize.table()`
`search_alg` is deprecated, in favor of [index_dist_type](https://github.com/tembo-io/pg_vectorize/blob/9e013e3c9c5702c982ca74415ff396074a744cb7/extension/src/api.rs#L22). Drop search_alg from the parameter and downstream call tree.

### PR DESCRIPTION
**Pull request by Bhavyajain21**
/fixes #146 
/claim #146

### CODE REVIEW

**Comment by ChuckHend** [comment]
`vectorize.search()` and `vectorize.table()` need to remain functional after the deprecated params are dropped, so there could be some scope related to updating the call tree such that it only uses the `index_dist_type` parameter.

**Comment by Bhavyajain21** [comment]
I've updated the PR. Can you please review? @ChuckHend

File: extension/sql/vectorize--0.13.0--0.14.0.sql [hunk 1]
```diff
@@ -45,9 +45,8 @@ CREATE  FUNCTION vectorize."init_rag"(
 	"unique_record_id" TEXT, /* &str */
 	"column" TEXT, /* &str */
 	"schema" TEXT DEFAULT 'public', /* &str */
-  "index_dist_type" TEXT DEFAULT 'pgv_hsnw_cosine', /* vectorize::types::IndexDist */
```
Comment (ChuckHend) [inline extension/sql/vectorize--0.13.0--0.14.0.sql:48]: Do not change any of the existing migration files, such as `extension/sql/vectorize--0.13.0--0.14.0.sql`. These files tell Postgres how to update the extension across specific versions. The changes in this PR will be coming in the next release, so there will need to be a new migration created `extension/sql/vectorize--0.19.0--0.20.0.sql`. In that migration file we will need to `DROP FUNCTION` for any functions that have their signature changed, then add the new `CREATE FUNCTION` definition.

**Comment by ChuckHend** [comment]
FYI - image builds will fail because the branch is outside the Tembo org, and that is fine.

File: extension/sql/vectorize--0.13.0--0.14.0.sql (hunk 1 above)
Comment (Bhavyajain21) [reply extension/sql/vectorize--0.13.0--0.14.0.sql]: Added new migration file

**Review by ChuckHend — APPROVED** [review]

**Pull request merged**